### ISSUE TITLE
Critical dependency: the request of a dependency is an expression

### ISSUE THREAD

**Issue by delasteve** [angular/angular-cli#2359]
1. OS?  Windows 7, 8 or 10. Linux (which distribution). Mac OSX (Yosemite? El Capitan?)
macOS El Capitan

2. Versions. Please run `ng --version`. If there's nothing outputted, please run in a Terminal: `node --version` and paste the result here:
   ```
   angular-cli: 1.0.0-beta.15
   node: 6.4.0
   os: darwin x64
   ```

3. Repro steps. Was this an app that wasn't created using the CLI? What change did you
   do on your code? etc.
`ng new testapp`
`cd testapp`
`ng test`

4. The log given by the failure. Normally this include a stack trace and some
   more information.
   ```
   WARNING in ./~/@angular/core/src/linker/system_js_ng_module_factory_loader.js
   45:15 Critical dependency: the request of a dependency is an expression

   WARNING in ./~/@angular/core/src/linker/system_js_ng_module_factory_loader.js
   57:15 Critical dependency: the request of a dependency is an expression
   ```

5. Mention any other details that might be useful.
I believe this is the fix for it. https://github.com/angular/angular/issues/11580#issuecomment-246880731
The warning only happens while running `ng test` but not when doing `ng build` or `ng serve`

**Comment by TheLarkInn** [comment]
Need to add ContextReplacementPlugin in test configs. A PR would be welcomed for this.

**Comment by JohannesRudolph** [comment]
I can confirm this fixes #2362 but it does not fix #2364

**Comment by aleury** [comment]
I think I may have a fix for this, which is pretty much as explained in the link provided above. May I submit a PR for this if no one else has claimed it @TheLarkInn ? Thanks!
